File: lecture_009(Recursion)/l_007_printzigzag.cpp
#include<iostream>
using namespace std;

void pzz(int n) {
    if(n==0) {
        return;
    }
    cout<<"Pre "<<n<<endl;
    pzz(n-1);
    cout<<"In "<<n<<endl;
    pzz(n-1);
    cout<<"Post "<<n<<endl;
}
int main() {
    int n;  
    cin>>n;
    pzz(n);

    return 0;
}

File: lecture_009(Recursion)/l_007_printzigzag.java
import java.util.Scanner;
public class l_007_printzigzag {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        int n = scn.nextInt();
        pzz(n);
    }
    public static void pzz(int n) {
        if(n==0) {
            return;
        }
        System.out.println("Pre "+n);
        pzz(n-1);
        System.out.println("In "+n);
        pzz(n-1);
        System.out.println("Post "+n);
    }
}